pages/alert/runtime-content.page.tsx

```diff
@@ -0,0 +1,130 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useContext, useState } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+
+import {
+  Alert,
+  AlertProps,
+  Box,
+  Button,
+  Checkbox,
+  ExpandableSection,
+  FormField,
+  Select,
+  SpaceBetween,
+} from '~components';
+import awsuiPlugins from '~components/internal/plugins';
+
+import AppContext, { AppContextType } from '../app/app-context';
+import ScreenshotArea from '../utils/screenshot-area';
+
+type PageContext = React.Context<AppContextType<{ loading: boolean; hidden: boolean; type: AlertProps.Type }>>;
+
+awsuiPlugins.alertContent.registerContentReplacer({
+  id: 'awsui/alert-test-action',
+  runReplacer(context, registerReplacement) {
+    console.log('mount');
+
+    const doReplace = () => {
+      registerReplacement('header', { type: 'original' });
+      registerReplacement('content', { type: 'original' });
+      if (context.type === 'error' && context.contentRef.current?.textContent?.match('Access denied')) {
+        registerReplacement('header', { type: 'remove' });
+        registerReplacement('content', {
+          type: 'replace',
+          onReplace: container => {
+            console.log('render replacement content');
+            render(
+              <SpaceBetween size="s">
+                <Box>---REPLACEMENT--- Access denied message! ---REPLACEMENT---</Box>
+                <ExpandableSection headerText="Original message">
+                  {context.contentRef.current?.textContent}
+                </ExpandableSection>
+              </SpaceBetween>,
+              container
+            );
+          },
+        });
+      }
+    };
+
+    doReplace();
+
+    return {
+      update() {
+        console.log('update');
+        doReplace();
+      },
+      unmount({ replacementContentContainer }) {
+        console.log('unmount');
+        unmountComponentAtNode(replacementContentContainer);
+      },
+    };
+  },
+});
+
+const alertTypeOptions = ['error', 'warning', 'info', 'success'].map(type => ({ value: type }));
+
+export default function () {
+  const {
+    urlParams: { loading = false, hidden = false, type = 'error' },
+    setUrlParams,
+  } = useContext(AppContext as PageContext);
+  const [unrelatedState, setUnrelatedState] = useState(false);
```

Comment:
This unrelated state does not demonstrate the any issues, because all alert content is primitive strings. Try placing JSX in the first alert and see what happens

Reply:
I added content swap to test page. When set - the content between the first and the second alert swaps. I observed no issues with this.

Reply:
I provided specific steps to reproduce the issue:

> Try placing JSX in the first alert and see what happens

You did not follow these steps and report you do not see the issue. 

I am not sure what else I can do here. I surrender on this thread.

Reply:
Please clarify what is the issue I should see if placing JSX. The replacement is already JSX. If I use some `<Box>content</Box>` instead of `'content'` that seems to work alright.

Reply:
This is the issue

https://github.com/user-attachments/assets/5f9ba540-bc2d-4755-a67f-8a1f8da828f3

I would say changing the "unrelated state" should not trigger any updates, none of these console logs should be happening.

If you think it is fine, handling possible impact of this risk is on you.

Reply:
Thanks, it is clear now. I think this is inline with how React works. If the JSX is memoized with memo or useMemo - the behaviour is not observed.

Reply:
The consuming team may put some expensive code inside the `doRender` function. The quip proposal mentions there might be a network call there. Are we okay with that many network calls on unrelated state changes?

Reply:
> If the JSX is memoized with memo or useMemo - the behaviour is not observed.

This means all existing `Alert` usage will have to be changed. I do not think it is a reasonable mitigation